nifi-nar-bundles/nifi-influxdb-bundle/nifi-influxdb-processors/src/main/java/org/apache/nifi/processors/influxdb/AbstractInfluxDBProcessor.java

```diff
@@ -0,0 +1,142 @@
+            .addValidator(StandardValidators.CHARACTER_SET_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor INFLUX_DB_URL = new PropertyDescriptor.Builder()
+            .name("influxdb-url")
+            .displayName("InfluxDB connection url")
+            .description("InfluxDB url to connect to")
+            .required(true)
+            .addValidator(StandardValidators.URL_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor DB_NAME = new PropertyDescriptor.Builder()
+            .name("influxdb-dbname")
+            .displayName("DB Name")
+            .description("InfluxDB database to connect to")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor USERNAME = new PropertyDescriptor.Builder()
+            .name("influxdb-username")
+            .displayName("Username")
+            .required(false)
+            .description("Username for accessing InfluxDB")
+            .addValidator(Validator.VALID)
+            .build();
+
+    public static final PropertyDescriptor PASSWORD = new PropertyDescriptor.Builder()
+            .name("influxdb-password")
+            .displayName("Password")
+            .description("Password for user")
+            .addValidator(Validator.VALID)
+            .sensitive(true)
+            .build();
+
+    protected static final PropertyDescriptor MAX_RECORDS_SIZE = new PropertyDescriptor.Builder()
+            .name("influxdb-max-records-size")
+            .displayName("Max size of records")
+            .description("Maximum size of records allowed to be posted in one batch")
+            .defaultValue("1 MB")
+            .required(true)
+            .addValidator(StandardValidators.DATA_SIZE_VALIDATOR)
+            .build();
+
+    static final Relationship REL_SUCCESS = new Relationship.Builder().name("success")
+            .description("Sucessful FlowFiles are routed to this relationship").build();
+
+    static final Relationship REL_FAILURE = new Relationship.Builder().name("failure")
+            .description("Failed FlowFiles are routed to this relationship").build();
+
+    public static final String INFLUX_DB_ERROR_MESSAGE = "influxdb.error.message";
+
+    protected InfluxDB influxDB;
+    protected String influxDbUrl;
+    protected String username;
+    protected String password;
+    protected long maxRecordsSize;
+
+    /**
+     * Helper method to help testability
+     * @return InfluxDB instance
+     */
+    protected InfluxDB getInfluxDB() {
+        return influxDB;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        username = context.getProperty(USERNAME).getValue();
+        password = context.getProperty(PASSWORD).getValue();
+        influxDbUrl = context.getProperty(INFLUX_DB_URL).getValue();
```

Comment:
I think it would be a better practice to not have these values stored as fields on the class. They get reset every time the user stops and restarts the processor, so I don't think we need to keep them around after the connection is established.